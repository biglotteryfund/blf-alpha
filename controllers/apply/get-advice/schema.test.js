/* eslint-env jest */
// @ts-nocheck
'use strict';

const schema = require('./schema');

function validate(data) {
    const { value, error } = schema.validate(data, {
        abortEarly: false,
        stripUnknown: true
    });

    return { value, error };
}

test('require one or more countries to be selected', () => {
    const validSingle = validate({
        country: 'england'
    });

    expect(validSingle.error).toBeNull();
    expect(validSingle.value).toEqual({ country: ['england'] });

    const validMultiple = validate({
        country: ['england', 'scotland']
    });

    expect(validMultiple.error).toBeNull();
    expect(validMultiple.value).toEqual({ country: ['england', 'scotland'] });

    expect(
        validate({
            country: 'invalid country'
        }).error.message
    ).toContain('"country" must be one of');
});
