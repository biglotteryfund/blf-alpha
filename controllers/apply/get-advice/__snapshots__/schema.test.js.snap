// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minimal invalid form 1`] = `
Array [
  "\\"projectCountry\\" must be one of [england, scotland, northern-ireland, wales]",
  "\\"projectLocation\\" must be a string",
  "\\"projectCosts\\" must be larger than or equal to 10000",
  "\\"projectDurationYears\\" must be less than or equal to 5",
  "\\"projectIdea\\" is required",
  "\\"organisationLegalName\\" is required",
  "\\"organisationAddress\\" is required",
  "\\"organisationType\\" is required",
  "\\"organisationBackground\\" is required",
  "\\"contactName\\" is required",
  "\\"contactEmail\\" is required",
]
`;

exports[`minimal valid form 1`] = `
Object {
  "contactEmail": "general.enquiries@tnlcommunityfund.org.uk",
  "contactName": Object {
    "firstName": "Björk",
    "lastName": "Guðmundsdóttir",
  },
  "contactPhone": "0345 410 2030",
  "organisationAddress": Object {
    "county": "West Midlands",
    "line1": "1234 example street",
    "postcode": "B15 1TR",
    "townCity": "Birmingham",
  },
  "organisationBackground": "Turnip greens yarrow ricebean rutabaga  endive cauliflower sea lettuce kohlrabi amaranth water spinach avocado daikon napa cabbage asparagus winter purslane kale. Celery potato scallion desert raisin horseradish spinach carrot soko. Lotus root water spinach fennel kombu maize bamboo shoot green bean swiss chard seakale pumpkin onion chickpea gram corn pea. Brussels sprout coriander water chestnut gourd swiss chard wakame kohlrabi beetroot carrot watercress. Corn amaranth salsify bunya nuts nori azuki bean chickweed potato bell pepper artichoke.",
  "organisationLegalName": "Example organisation",
  "organisationTradingName": "Example trading name",
  "organisationType": "Social enterprise",
  "projectCosts": 250000,
  "projectCountry": Array [
    "england",
  ],
  "projectDurationYears": 3,
  "projectIdea": "Turnip greens yarrow ricebean rutabaga  endive cauliflower sea lettuce kohlrabi amaranth water spinach avocado daikon napa cabbage asparagus winter purslane kale. Celery potato scallion desert raisin horseradish spinach carrot soko. Lotus root water spinach fennel kombu maize bamboo shoot green bean swiss chard seakale pumpkin onion chickpea gram corn pea. Brussels sprout coriander water chestnut gourd swiss chard wakame kohlrabi beetroot carrot watercress. Corn amaranth salsify bunya nuts nori azuki bean chickweed potato bell pepper artichoke.",
  "projectLocation": "placeholder-location",
  "projectLocationDescription": "optional description",
}
`;
